### ISSUE TITLE
Documentar generación de la ISO en la wiki

### ISSUE THREAD

**Issue by jasvazquez** [aosucas499/minino-TDE#49]
Todavía no me he decidido a crear una ISO pues tendría que ponerme a mirar cómo lo está haciendo @aosucas499.

Entiendo que no será complejo y que seguramente exista algo de documentación hecha (aunque sea en inglés) que se podría aprovechar (ahorrándonos trabajo de documentación) pero sería interesante contar con una pequeña reseña (tipo resumen) que facilite la tarea la primera vez 😅

**Comment by aosucas499** [comment]
No le he dedicado tiempo al instalador, debido a que me he centrado en el problema de las pizarras en educaandos y ahora a instalar minino en la tablet vexia, pero la última vez que probé el instalador, los post-script no funcionaron, lo utilicé para borrar el acceso directo del escritorio en la versión instalada, pero nada, quizás necesitaría un chroot. Solo había que añadir scripts a esa carpeta de post-script en la carpeta de refracta installer.

 Aunque no le he podido dedicar tiempo creo que al final tendremos que borrar el refracta installer y los script de installer de minino que se basa en refracta y construir los nuestros. Pues al no haberlos investigado,  vamos a ciegas.

En el update-minino, están al final los dos comandos necesarios para crear la ISO (comentados #) tal cual dejas el sistema, no hay dificultades. El problema es que no lo hemos investigado y quizás por eso algunos métodos de instalación dan fallo. En el script update-minino están las carpetas de refracta y algunas cosas que modifiqué del instalador, como el grub o algún post-script que no funcionó. Puede fallar cualquier cosa, necesitamos reestructurar refracta a nuestro modo, desinstalarlo, borrar el minino installer y construir el nuestro, por ejemplo, aunque el sistema está preparado, puede fallar que hemos cambiado el usuario a usuario y en la versión original era minino.

Espero tener más tiempo en breve y que le podamos dar una solución al instalador que tantos problemas está dando, así como incluir tu código para esas tarjetas de sonido.

**Comment by aosucas499** [comment]
Lo primero que voy a probar es a instalar una versión similar de refracta installer a la que traía el sistema, la última que vez puse una más nueva y puede que eso traiga problemas, voy a probar con la versión que traía minino original. También probaré a dejar el usuario minino, a ver qué ocurre.

**Comment by aosucas499** [comment]
He visto que no borré refractainstaller al cambiar por systemback, solo borré refractasnapshot que es el que hace la ISO. Poco problema daba esto entonces. Es el encargado de construir la ISO y la ISO la hace bien, sigue habiendo problema con refracta installer y siempre lo hubo, por eso cambié a Systemback.

**Comment by jasvazquez** [comment]
Uufff... mucha incertidumbre veo en este punto joven padawan 🤣👍🏻

**Comment by aosucas499** [comment]
Me he liado esta mañana, dormí poco jajajaja. Igualmente voy a probar todas las cosas hasta que tengamos un instalador decente. ¿Qué opinas tú sobre el instalador? ¿Debería seguir habiendo opciones y diferentes formas o solo una desatendida y sin muchas opciones debido a que se suele instalar en equipos que no necesitan mucho uso de particiones?

**Comment by jasvazquez** [comment]
Hombre, tener posibilidad de elegir (aunque nos complique la vida) siempre es un punto a su favor pero he de confesar que he estado tentado de plantear una instalación desatendida que se cargue todas las particiones e instale (sí o sí) Minino en el equipo 😅 

Un poco radical pero, como decían en el Milagro de P. Tinto: si hay que sanear, se sanea 🤣 

![imagen](https://user-images.githubusercontent.com/604744/105634248-67b7cb80-5e5d-11eb-9061-84af5759aaf5.png)

**Comment by jasvazquez** [comment]
Dejo aquí referencia a la escasa documentación que existe (de forma oficial) sobre [Refracta](https://refracta.org/documents.html) para cuando abordemos este tema

**Comment by aosucas499** [comment]
Interesante tener aquí el código del archivo minino-installer-selector, alojado en /usr/local/bin. Este archivo lleva al tipo de instalación que hemos seleccionado.

![VirtualBox_minino TDE_24_01_2021_18_26_12](https://user-images.githubusercontent.com/72696244/105638202-b3747000-5e71-11eb-97bc-2393a5ca494f.png)

**Comment by aosucas499** [comment]
Como vemos hay diferentes instalaciones: 

a) desasistido: usa el archivo **minino-installer-b** alojado en esta misma carpeta (/usr/local/bin).

b) básico: utiliza **remastersys-installer**, alojado en /usr/bin

c) avanzado: utilizar el archivo **minino-installer-r**, alojado en /usr/bin.

d) experto: utilizar el archivo **refractainstaller-e-wrapper.sh**, alojado en /usr/bin.

**Comment by aosucas499** [comment]
No veo mala idea borrar estos casos del instalador y dejar solo la desasistida, es la más efectiva, incluso es la única que me ha detectado las particiones de la tablet vexia que usa tarjeta de memoria en lugar de disco duro "/dev/mmcblk0". En la versión manual de esta opción incluso te deja seleccionar y editar las particiones con gparted. Siempre podemos solo esconder la opción sabiendo que las tenemos ahí disponibles en caso de querer usarlas para probar o solucionar alguna instalación. ¿Qué opinas?

**Comment by aosucas499** [comment]
Otra de las cosas que teníamos pendiente y que más "morcilla" da a los compañeros, es que la instalación desasistida, hay veces que confunde los dispositivos del pendrive de minino live con el que arrancamos y el disco duro a instalar. Parece ser que lo instala en el usb otra vez y los compañeros se encuentran con un sistema tal cual estaba antes de la instalación.

Esta es la función que detecta el sistema "live" y asigna valores "sda o sdb". La voy a estudiar, a ver si ves el fallo.

![VirtualBox_minino TDE_24_01_2021_18_54_22](https://user-images.githubusercontent.com/72696244/105639024-35669800-5e76-11eb-8ae4-73ae1a8eb3a3.png)

No estoy a estas horas muy espabilado hoy, pero ¿no está al revés?

**Comment by jasvazquez** [comment]
No, está mirando en qué partición se ha montado _/live/image_ y se asegura que no sea _/dev/sda1_.
Si no lo es, utiliza _/dev/sda1_ como destino de la instalación (y si resulta que en _/dev/sda1_ está montado el _/live/image_, se va por la otra unidad)

No me parece muy limpio lo que está haciendo y presupone cosas que no debería, por lo que no me extraña que falle.

Sería interesante conocer qué valor tiene el resultado de ejecutar `mount | grep "/live/image" | cut -d " " -f 1` en alguno de los equipos en los que da problemas

**Comment by aosucas499** [comment]
Haciendo búsquedas en los cuatro archivos de la instalación (o alguno al que se refiere/conecta alguno de ellos) observo que las opciones desatendida o básica no lanzan los pre-scripts o post-scripts de instalación. Sí que hacen referencia en su código, las instalaciones avanzada y experto. Quizás por eso falló mi intento de borrar el icono del escritorio, por impaciente claramente.

![VirtualBox_minino TDE_24_01_2021_19_17_56](https://user-images.githubusercontent.com/72696244/105639530-31884500-5e79-11eb-8d7e-adc6a0efe211.png)

![VirtualBox_minino TDE_24_01_2021_19_18_12](https://user-images.githubusercontent.com/72696244/105639531-351bcc00-5e79-11eb-8be0-5cfe0feda769.png)
